File: src/Lykke.Service.PayInvoicePortal/Models/Refund/RefundDataResponse.cs
using System.Collections.Generic;
using Lykke.Service.PayInvoicePortal.Models.Assets;

namespace Lykke.Service.PayInvoicePortal.Models.Refund
{
    public class RefundDataResponse
    {
        public AssetModel PaymentAsset { get; set; }

        public IEnumerable<string> SourceWalletAddresses { get; set; }

        public decimal Amount { get; set; }
    }
}

File: src/Lykke.Service.PayInvoicePortal/wwwroot/ng2/src/app/components/InvoiceDetails/InvoiceDetailsRefundDialog/RefundDataResponse.ts
import { AssetModel } from 'src/app/models/AssetModel';

export class RefundDataResponse {
  paymentAsset: AssetModel;
  sourceWalletAddresses: string[] = [];
  amount: number;

  static isValid(res: RefundDataResponse): boolean {
    return res.paymentAsset && res.sourceWalletAddresses.length && res.amount > 0;
  }
}
